framework/db/ActiveRecord.php

```diff
@@ -171,20 +171,16 @@ protected static function findByCondition($condition)
     {
         $query = static::find();
 
-        if (!ArrayHelper::isAssociative($condition)) {
+        $isConditionArray = is_array($condition);
+        if (!$isConditionArray || ($isConditionArray && !ArrayHelper::isAssociative($condition) && count($condition) < 3)) {
```

Comment:
First of all, it could be simplified to:

```php
!$is_array($condition) || (!ArrayHelper::isAssociative($condition) && count($condition) < 3)
```

Then `< 3` looks very suspicious. There could be a composite primary key which consists of any number of columns, not necessary two. So the condition is wrong.